WordPress.gitignore

```diff
@@ -1,6 +1,10 @@
 # ignore everything in the root except the "wp-content" directory.
+/*
```

Comment:
> This file is missing `/*` to begin with ignoring everything. 

Are you able to share some context about why this is a safe change to make?

Reply:
Without /* the negated lines(like !wp-content) are irrelevant. This whole file is built on the concept that only the negated lines are to be included, for that to work everything must be excluded initially.

Reply:
Yep, @stkrzysiak is right. Right now part with "!wp-content/" don't work ;/ 
We need that "/*" . Otherwise files of core wordpress are now added to change

Reply:
I want to pick up on @shiftkey comment, namely about the _safety_ aspect of that specific `/*` rule—I assume that was something along this @shiftkey was referring to.

Wasn't better (and safer, more below) if instead of a catch-all approach like that, we should opt for a more **declarative** way of stating the folders we want to see ignored instead?

I'm concerned that using the 'catch-all' approach we might raise some conflict with other .gitignores, or hinder if one wants to mix-and-match several of these .gitignore files like on [gitignore.io](http://gitignore.io) and some technology requires some folder on the root to be added that would be caught by this rule, etc.

Does this makes sense?

Reply:
FYI, added the #3295 PR that addresses my concern above.

Reply:
I will redo this to change to previous to enable this pull request to be merge and close some issues. 
This need to be fixed in #3295 with another pull request after all paths are declarative and battle tested for errors
Thank you @pedrosanta for helping understand better context

Reply:
Declarative way is better imho. Need to be discuss on approach and tests

Reply:
Yes #3295 has a declarative approach, I based it mostly on folders from a fresh WP install, and tested it, but a couple common folders on WP ecosystem might or nor be missing, but then again I have the strong feeling we should only add into **.gitignore** what is generally assumed as widely as possible as a common practice/setup on WP ecosystem. Because of course YMMV and it's supposed for you to branch of the .gitignore, and then add rules for your particular setup.